### ISSUE TITLE
ORT should fail with missing location parameters

### ISSUE THREAD

**Issue by jhg03a** [apache/trafficcontrol#3970]
## I'm submitting a ...

- [ ] bug report
- [ ] new feature / enhancement request
- [X] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [X] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

If there is a missing mid header rewrite location parameter somehow that's referenced by the ATS remap config, ORT is ok with this and ATS just fails to reload properly.

## Expected / new behavior:

ORT should fail if location parameters are missing.

**Comment by ocket8888** [comment]
Seems like that ought to fail at the config generation step; wherever that's occurring these days.

**Comment by jhg03a** [comment]
Header rewrite files and remap.config are considered separate files during config generation.

**Comment by ocket8888** [comment]
Oh, I misunderstood. I thought you were saying there was something wrong syntactically with the file's contents. _Parameter_ meaning **ATC** Parameter, got it.

**Comment by mitchell852** [comment]
@rob05c - is this still relevant with all the ort/atstccfg changes?

**Comment by rob05c** [comment]
Yeah, we still use location parameters, and ORT/config gen still "succeeds" even if they're missing.

We need to just get rid of DS location params, and infer them. But it doesn't yet.

**Comment by rob05c** [comment]
>Seems like that ought to fail at the config generation step

Yeah, I made the ORT Config Gen emulate the old as much as possible, for safety.

I think it should log an Error, but keep going, for this kind of thing. We shouldn't break everything for one bad DS, and we should always try to keep the CDN up. But it should definitely log an error, and people should make alarms for that.
